addons/account/models/account_account.py

```diff
@@ -374,14 +375,19 @@ def _compute_opening_debit_credit(self):
     def _compute_account_type(self):
         """ Compute the account type based on the account code.
         Search for the closest parent account code and sets the account type according to the parent.
+        If there is no parent (e.g. the account code is lower than any other existing account code),
+        the account type will be set to 'asset_current'.
         """
         accounts_to_process = self.filtered(lambda r: r.code and not r.account_type)
-        accounts_with_codes = {r['code']: r['account_type'] for r in self.search_read(domain=[(
-            'code', '!=', False), ('company_id', '=', self.company_id.id)], fields=['code', 'account_type'])}
-        codes_list = list(accounts_with_codes.keys())
+        all_accounts = self.search_read(domain=[('code', '!=', False), ('company_id', 'in', self.company_id.ids)], fields=['code', 'account_type', 'company_id'])
```

Comment:
We only need those from companies of `accounts_to_process` which might be a subset
```suggestion
        all_accounts = self.search_read(domain=[('code', '!=', False), ('company_id', 'in', accounts_to_process.company_id.ids)], fields=['code', 'account_type', 'company_id'])
```